src/main/java/net/mcreator/element/converter/v2023_4/RangedItemToProjectileAndItemConverter.java

```diff
@@ -0,0 +1,182 @@
+			}
+
+			if (rangedItem.get("onBulletHitsBlock") != null)
+				projectile.onHitsBlock = new Procedure(
+						rangedItem.get("onBulletHitsBlock").getAsJsonObject().get("name").getAsString());
+			if (rangedItem.get("onBulletHitsEntity") != null)
+				projectile.onHitsEntity = new Procedure(
+						rangedItem.get("onBulletHitsEntity").getAsJsonObject().get("name").getAsString());
+			if (rangedItem.get("onBulletHitsPlayer") != null)
+				projectile.onHitsPlayer = new Procedure(
+						rangedItem.get("onBulletHitsPlayer").getAsJsonObject().get("name").getAsString());
+			if (rangedItem.get("onBulletFlyingTick") != null)
+				projectile.onFlyingTick = new Procedure(
+						rangedItem.get("onBulletFlyingTick").getAsJsonObject().get("name").getAsString());
+
+			projectile.getModElement()
+					.setParentFolder(FolderElement.dummyFromPath(input.getModElement().getFolderPath()));
+			workspace.getModElementManager().storeModElementPicture(projectile);
+			workspace.addModElement(projectile.getModElement());
+			workspace.getGenerator().generateElement(projectile);
+			workspace.getModElementManager().storeModElement(projectile);
+
+			Item item = new Item(new ModElement(workspace, input.getModElement().getName(), ModElementType.ITEM));
+			item.name = rangedItem.get("name").getAsString();
+			item.texture = rangedItem.get("texture").getAsString();
+			item.renderType = rangedItem.get("renderType").getAsInt();
+			if (rangedItem.get("customModelName") != null)
+				item.customModelName = rangedItem.get("customModelName").getAsString();
+			if (item.customModelName.equals("Normal") && item.renderType == 0)
+				item.customModelName = "Ranged item"; // Convert the ranged items' normal model to items' RI model option
+			item.creativeTab = new TabEntry(workspace,
+					rangedItem.get("creativeTab").getAsJsonObject().get("value").getAsString());
+			List<String> infoFixedValues = new ArrayList<>();
+			String infoProcedureName = null;
+
+			if (rangedItem.get("specialInfo") != null) {
+				rangedItem.getAsJsonArray("specialInfo").iterator()
+						.forEachRemaining(element -> infoFixedValues.add(element.getAsString()));
+			} else if (rangedItem.get("specialInformation") != null) {
+				if (rangedItem.get("specialInformation").getAsJsonObject().get("fixedValue") != null) {
+					rangedItem.get("specialInformation").getAsJsonObject().getAsJsonArray("fixedValue").iterator()
+							.forEachRemaining(element -> infoFixedValues.add(element.getAsString()));
+				}
+				infoProcedureName = rangedItem.get("specialInformation").getAsJsonObject().get("name").getAsString();
+			}
+			item.specialInformation = new StringListProcedure(infoProcedureName, infoFixedValues);
+
+			item.stackSize = rangedItem.get("stackSize").getAsInt();
+			if (rangedItem.get("animation") != null)
+				item.animation = rangedItem.get("animation").getAsString();
+			else
+				item.animation = "bow";
+
+			if (rangedItem.has("glowCondition"))

```

Comment:
You are checking for glowCondition here and then again in else if, there is something wrong here likely

Reply:
Oh, I forgot to remove it when changing the converter (to apply the glow feature converter you made)... but it never caused a problem during my tests 🤷

Reply:
I don't think it will cause problems, but it is still unclean code

Reply:
It didn't seem to cause a problem during my tests, so this is why I didn't notice my mistake 🫠

Reply:
I re-tested and this still works fine.